spec/features/create_una_book_spec.rb

```diff
@@ -0,0 +1,102 @@
+    Hyrax::PermissionTemplateAccess.create!(
+      permission_template_id: permission_template.id,
+      agent_type: "user",
+      agent_id: user.user_key,
+      access: "deposit"
+    )
+    login_as user
+    visit new_work_path
+  end
+
+  it 'renders the new UNA Work page' do
+    expect(page).to have_content "Add New Una Book"
+  end
+
+  it 'adds files to work' do
+    click_link "Files"
+    expect(page).to have_content "Add files"
+    expect(page).to have_content "Add folder"
+    within('span#addfiles') do
+      attach_file("files[]", Rails.root.join('spec', 'fixtures', 'hyrax', 'image.jp2'), visible: false)
+      attach_file("files[]", Rails.root.join('spec', 'fixtures', 'hyrax', 'jp2_fits.xml'), visible: false)
+    end
+  end
+
+  it 'applys work visibility' do
+    find('body').click
+    choose("#{work_type}_visibility_open")
+
+    expect(page).to have_content('Please note, making something visible to the world (i.e. marking this as Public) may be viewed as publishing which could impact your ability to')
+  end
+
+  it 'saves the work' do
+    click_link "Descriptions"
+    fill_in("#{work_type}_title", with: 'My Test Work')
+    select('Organisational', from: "#{work_type}_creator__creator_name_type")
+    fill_in("#{work_type}_creator__creator_organization_name", with: 'Ubiquity Press')
+    check('agreement')
+    click_on('Save')
+    expect(page).to have_content('My Test Work')
+    expect(page).to have_content('Public')
+    expect(page).to have_content("Your files are being processed by Hyku in the background.")
+  end
+
+  context "when rendering the form" do
+    before do
+      click_on "Additional fields"
+    end
+
+    it "renders all simple worktype fields" do
+      worktype_simple_fields = %w[title alt_title resource_type alt_email abstract
+                                  keyword subject org_unit language license version_number
+                                  table_of_contents publisher place_of_publication
+                                  edition isbn official_link series_name
+                                  location longitude latitude add_info buy_book]
+      worktype_simple_fields.each do |field|
+        expect(page).to have_field("#{work_type}_#{field}")
+      end
+    end
+
+    it "renders complex name fields" do
+      worktype_complex_name_fields = %w[creator contributor]
+      worktype_complex_name_fields.each do |field|
+        expect(page).to have_field("#{work_type}_#{field}__#{field}_family_name")
+        expect(page).to have_field("#{work_type}_#{field}__#{field}_given_name")
+      end
+    end
+
+    it "renders complex identifier fields" do
+      worktype_complex_fields = %w[related_identifier]
+      worktype_complex_fields.each do |field|
+        expect(page).to have_field("#{work_type}_#{field}__#{field}")
+      end
+    end
+
+    it "renders all date fields" do
+      worktype_date_fields = %w[date_published]
+      worktype_date_fields.each do |field|
```

Comment:
Can we get rid of these excess variables, they are not useful.

Reply:
They make it easier to copy the tests when making worktypes and when you have very long arrays they keep the tests more readable